frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/pipeline-details/PipelineVisualization.tsx

```diff
@@ -1,15 +1,33 @@
 import * as React from 'react';
-import { K8sResourceKind } from '@console/internal/module/k8s';
-import { getPipelineTasks } from '../../../../utils/pipeline-utils';
-import { PipelineVisualizationGraph } from './PipelineVisualizationGraph';
+import { Alert } from '@patternfly/react-core';
+import { Pipeline, PipelineRun } from '../../../../utils/pipeline-augment';
+import PipelineTopologyGraph from './pipeline-topology/PipelineTopologyGraph';
+import { getTopologyNodesEdges } from './pipeline-topology/utils';
 
-export interface PipelineVisualizationProps {
-  pipeline?: K8sResourceKind;
+interface PipelineTopologyVisualizationProps {
+  pipeline: Pipeline;
+  pipelineRun?: PipelineRun;
 }
 
-export const PipelineVisualization: React.FC<PipelineVisualizationProps> = ({ pipeline }) => (
-  <PipelineVisualizationGraph
-    namespace={pipeline.metadata.namespace}
-    graph={getPipelineTasks(pipeline)}
-  />
-);
+const PipelineVisualization: React.FC<PipelineTopologyVisualizationProps> = ({
+  pipeline,
+  pipelineRun,
+}) => {
+  const { nodes, edges } = getTopologyNodesEdges(pipeline, pipelineRun);
+
+  if (nodes.length === 0 && edges.length === 0) {
+    // Nothing to render
+    // TODO: Confirm wording with UX; ODC-1860
+    return <Alert variant="info" isInline title="This Pipeline has no tasks to visualize." />;
```

Comment:
@serenamarie125 I took a causal stab at solving https://issues.redhat.com/browse/ODC-1860 when I stumbled upon the issue in my work.